qa/README.md

```diff
@@ -80,25 +80,36 @@ public class ProcessDefinitionQueryTest {
 
 **Need:**
 
-* We need to configure the broker for specific authentication
-* We need to make use of the `@RegisterExtension` to pass in the Broker, pre-configured
-* We need to make sure that the test is tagged with `@Tag("multi-db-test")`
+* We need to configure the broker for specific authentication, for that we need to make use of `TestStandaloneBroker` or `TestSimpleCamundaApplication`
+* We need to annotate the test applications with `MultiDbTestApplication`, to mark application managed by `CamundaMultiDbExtension`
+
+> [!Note]
+>
+> The separation of test application and extension (to configure different secondary storages) is on purpose.
+>
+> This is for:
+>  * Single responsibility
+>    * the `CamundaMultiDbExtension` only focuses on configuring the database based on various factors (e.g. CI job)
+>  * KISS: to keep things light
+>  * Favor composability: make it reusable
+>    * the extension handles configuring the DB, and can be used in different occasions together with using the standalone Camunda, standalone Broker, or something different
+>    * the test applications focus on the respective application lifecycle and flexible configuration
```

Comment:
Based on feedback by @npepinpe https://github.com/camunda/camunda/pull/28911#discussion_r1978911431

Reply:
I reworked this here [d00c397](https://github.com/camunda/camunda/pull/29927/commits/d00c397506397c084750979ed6156ec3723dc184)